clients/fossid-webapp/src/main/kotlin/serviceExtensions.kt

```diff
@@ -0,0 +1,273 @@
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+@file:Suppress("TooManyFunctions")
+
+package org.ossreviewtoolkit.fossid
+
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+
+import kotlin.reflect.KClass
+
+import org.ossreviewtoolkit.model.jsonMapper
+
+import retrofit2.Call
+
+private const val SCAN_GROUP = "scans"
+private const val PROJECT_GROUP = "projects"
+
+/**
+ * This function verifies that a request to FossID server was successful
+ * @param operation the label of the operation, for logging
+ * @param withDataCheck if true, the payload will be checked for existence
+ */
+fun EntityPostResponseBody<*>?.checkResponse(operation: String, withDataCheck: Boolean = true) {
+    checkNotNull(this)
+    require(error == null) {
+        "Could not '$operation'. Additional information : $error"
+    }
+    if (withDataCheck) {
+        requireNotNull(data) {
+            "No Payload received for '$operation'. Additional information : $error"
+        }
+    }
+}
+
+/**
+ * The list operations in FossID have inconsistent return types depending on the amount of entities returned.
+ * This function streamlines these entities to a list.
+ */
+fun <T : Any> EntityPostResponseBody<Any>.toList(cls: KClass<T>): List<T> {
+    // the list  operation returns different json depending on the amount of scans
+    return when (val data = data) {
+        is List<*> -> {
+            data.map { jsonMapper.convertValue(it, cls.java) }
+        }
+        is Map<*, *> -> {
+            data.values.map { jsonMapper.convertValue(it, cls.java) }
+        }
+        is Boolean -> {
+            emptyList()
+        }
+        else -> {
+            error("Cannot process the returned values")
+        }
+    }
+}
+
+/**
+ * Get a FossID project
+ * @param apiKey the key to query the API
+ * @param user the user querying the API
+ * @param projectCode the code of the project
+ */
+fun FossIdRestService.getProject(apiKey: String, user: String, projectCode: String) = getProject(
+        PostRequestBody("get_information", PROJECT_GROUP, apiKey, user, "project_code" to projectCode)
+)
+
+/**
+ * List the scans of a FossID project
+ * @param apiKey the key to query the API
+ * @param user the user querying the API
+ * @param projectCode the code of the project
+ */
+fun FossIdRestService.listScansForProject(apiKey: String, user: String, projectCode: String) = listScansForProject(
+        PostRequestBody("get_all_scans", PROJECT_GROUP, apiKey, user, "project_code" to projectCode)
+)
+
+/**
+ * Create a new FossID project
+ * @param apiKey the key to query the API
+ * @param user the user querying the API
+ * @param projectCode the code of the project
+ * @param projectName the name of the project
+ * @param comment the comment of the project
+ */
+fun FossIdRestService.createProject(
+        apiKey: String,
```

Comment:
Use 4 spaces for indentation, not 8. Similar below.

Reply:
done